### ISSUE TITLE
Installation source was chosen - Vagrant

### ISSUE THREAD

**Issue by skloeckner** [ceph/ceph-ansible#1210]
Here's the error I'm getting once ansible tries to run on my vagrant environment:

~~~~
TASK [ceph.ceph-common : make sure an installation source was chosen] **********
fatal: [mon0]: FAILED! => {"changed": false, "failed": true, "msg": "choose an upstream installation source or read https://github.com/ceph/ceph-ansible/wiki"}
fatal: [mon1]: FAILED! => {"changed": false, "failed": true, "msg": "choose an upstream installation source or read https://github.com/ceph/ceph-ansible/wiki"}
fatal: [mon2]: FAILED! => {"changed": false, "failed": true, "msg": "choose an upstream installation source or read https://github.com/ceph/ceph-ansible/wiki"}
	to retry, use: --limit @/tmp/ceph-ansible/site.retry

PLAY RECAP *********************************************************************
mon0                       : ok=2    changed=0    unreachable=0    failed=1   
mon1                       : ok=2    changed=0    unreachable=0    failed=1   
mon2                       : ok=2    changed=0    unreachable=0    failed=1   
osd0                       : ok=2    changed=0    unreachable=0    failed=0   
osd1                       : ok=2    changed=0    unreachable=0    failed=0   
osd2                       : ok=2    changed=0    unreachable=0    failed=0   

Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
~~~~

Here's my vagrant_variables.yml:

~~~~
---

# DEPLOY CONTAINERIZED DAEMONS
docker: false

# DEFINE THE NUMBER OF VMS TO RUN
mon_vms: 3
osd_vms: 3
mds_vms: 0
rgw_vms: 0
nfs_vms: 0
rbd_mirror_vms: 0
client_vms: 0
iscsi_gw_vms: 0

# Deploy RESTAPI on each of the Monitors
restapi: true

# SUBNETS TO USE FOR THE VMS
public_subnet: 192.168.42
cluster_subnet: 192.168.43

# MEMORY
# set 1024 for CentOS
memory: 512

# Ethernet interface name
# use eth1 for libvirt and ubuntu precise, enp0s8 for CentOS and ubuntu xenial
eth: 'eth1'

# Disks
# For Xenial use disks: "[ '/dev/sdb', '/dev/sdc' ]"
# For CentOS7 use disks: "[ '/dev/sda', '/dev/sdb' ]"
disks: "[ '/dev/sdb', '/dev/sdc' ]"

# VAGRANT BOX
# Ceph boxes are *strongly* suggested. They are under better control and will
# not get updated frequently unless required for build systems. These are (for
# now):
#
# * ceph/ubuntu-xenial
#
# Ubuntu: ceph/ubuntu-xenial or bento/ubuntu-16.04 or ubuntu/trusty64 or ubuntu/wily64
# CentOS: bento/centos-7.1 or puppetlabs/centos-7.0-64-puppet
# libvirt CentOS: centos/7
# parallels Ubuntu: parallels/ubuntu-14.04
# Debian: deb/jessie-amd64 - be careful the storage controller is named 'SATA Controller'
# For more boxes have a look at:
#   - https://atlas.hashicorp.com/boxes/search?utf8=✓&sort=&provider=virtualbox&q=
#   - https://download.gluster.org/pub/gluster/purpleidea/vagrant/
vagrant_box: bento/ubuntu-14.04
# The sync directory changes based on vagrant box
# Set to /home/vagrant/sync for Centos/7, /home/{ user }/vagrant for openstack and defaults to /vagrant
vagrant_sync_dir: /home/vagrant/sync
# VAGRANT URL
# This is a URL to download an image from an alternate location.  vagrant_box
# above should be set to the filename of the image.
# Fedora virtualbox: https://download.fedoraproject.org/pub/fedora/linux/releases/22/Cloud/x86_64/Images/Fedora-Cloud-Base-Vagrant-22-20150521.x86_64.vagrant-virtualbox.box
# Fedora libvirt: https://download.fedoraproject.org/pub/fedora/linux/releases/22/Cloud/x86_64/Images/Fedora-Cloud-Base-Vagrant-22-20150521.x86_64.vagrant-libvirt.box
# vagrant_box_url: https://download.fedoraproject.org/pub/fedora/linux/releases/22/Cloud/x86_64/Images/Fedora-Cloud-Base-Vagrant-22-20150521.x86_64.vagrant-virtualbox.box

os_tuning_params:
  - { name: kernel.pid_max, value: 4194303 }
  - { name: fs.file-max, value: 26234859 }
~~~~

I even uncommented out group_vars/all.yml:

~~~~
# STABLE
########

 COMMUNITY VERSION
ceph_stable: false # use ceph stable branch
ceph_stable_key: https://download.ceph.com/keys/release.asc
ceph_stable_release: jewel # ceph stable release
ceph_stable_repo: "http://download.ceph.com/debian-{{ ceph_stable_release }}"
~~~~

My experience with this ansible playbook:
I've used 1.0.8 to test a vagrant environment in virtualbox to the point where I felt comfortable using it to deploy a production environment. Turns out I need it again in a vagrant environment to make some sweeping changes and I wanted to test out my current config in a vagrant environment before making the change live.

In the past, I was able to just follow the basic steps for copying the vagrant_variables.yml and site.yml and group_vars/all.yml files from the samples and all I needed to change was the vagrant_variables.yml file in order to get a vagrant environment up and running.

Steps I've taken so far in troubleshooting:

1. Upgraded ansible from 2.1 to 2.2
2. Uncommented the group_vars/all.yml file for the community version which was only necessary for my production deployment, vagrant never needed this(although it did not work).
3. Tried with the latest release.
4. Tried with cloning the latest commit "git clone https://github.com/ceph/ceph-ansible.git"

I wanted to submit this as it appears to be a bug but I'm not quite sure where the bug is because there is no selection I need to make in the vagrant_variables.yml file as far as a software repository is concerned, it's always been assumed it would just take from the latest stable, which is jewel.

**Comment by skloeckner** [comment]
Well, this is embarrassing. Shortly after writing this, I realized my own problem and that I left this as the defaults

~~~~
ceph_stable: false # use ceph stable branch
~~~~

I just changed it to true and it's running now. Will close once I can confirm that completes what's now necessary for the vagrant steps.

Perhaps this should be added to the readme in order to emphasize the extra step? It doesn't seem like the vagrant steps work anymore.
